pkg/binding/bindingpolicy-resolution.go

```diff
@@ -36,8 +37,10 @@ import (
 type bindingPolicyResolution struct {
 	sync.RWMutex
 
-	objectIdentifierToResourceVersion map[util.ObjectIdentifier]string
-	destinations                      sets.Set[string]
+	objectIdentifierToResourceVersion  map[util.ObjectIdentifier]string
+	objectIdentifierToStatusCollectors map[util.ObjectIdentifier]sets.Set[string]
```

Comment:
I think the code would be simpler and more efficient if these two maps were merged into one, something like the following.

```
objectToData map[util.ObjectIdentifier] objectData
```

where

```
type objectData struct {ResourceVersion string; StatusCollectors sets.Set[string]}
```

Reply:
done